pkg/apis/apps/validation/validation.go

```diff
@@ -69,6 +69,30 @@ func ValidatePodTemplateSpecForStatefulSet(template *api.PodTemplateSpec, select
 	return allErrs
 }
 
+func ValidatePersistentVolumeClaimDeletePolicy(policy apps.PersistentVolumeClaimDeletePolicyType, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	switch {
+	case policy == apps.RetainPersistentVolumeClaimDeletePolicyType:
+	case policy == apps.DeletePersistentVolumeClaimDeletePolicyType:
+	default:
+		allErrs = append(allErrs, field.NotSupported(fldPath, policy, []string{string(apps.RetainPersistentVolumeClaimDeletePolicyType), string(apps.DeletePersistentVolumeClaimDeletePolicyType)}))
+	}
+	return allErrs
+}
+
+func ValidatePersistentVolumeClaimPolicy(policy *apps.StatefulSetPersistentVolumeClaimPolicy, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) {
```

Comment:
Not sure about the recommendation here.
Will leave it to @liggitt

Reply:
I think it is fair to check policy != nil if feature is enabled (although not sure if we do this anywhere else), but I say we should always validate the internals if policy != nil, regardless of the gate.

Reply:
Let me actually reconsider this point. Since we default to retain, and that is the common use-case for statefulsets, it seems like the UX might be nicer to let this policy be optional?

That is to say we shouldn't !nil-check. WDYT?

Validating regardless of the gate makes sense.

Reply:
> That is to say we shouldn't !nil-check. WDYT?

At least that's what we did for other recent API changes. So, SGTM.

Reply:
Done.

I didn't have to change the validation tests, so I guess this is what we were expecting to do anyway. Defaulting doesn't let the policy be nil when the featuregate is enabled.